data/src/main/java/com/microsoft/azure/kusto/data/auth/KnownKeywords.java

```diff
@@ -0,0 +1,72 @@
+package com.microsoft.azure.kusto.data.auth;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+public enum KnownKeywords {
+    DATA_SOURCE("Data Source", true),
+    INITIAL_CATALOG("Initial Catalog", true),
+    FEDERATED_SECURITY("AAD Federated Security", true),
+    APPLICATION_CLIENT_ID("Application Client Id", true),
+    APPLICATION_KEY("Application Key", true),
+    USER_ID("User ID", true),
+    AUTHORITY_ID("Authority Id", true),
+    APPLICATION_TOKEN("Application Token", true),
+    USER_TOKEN("User Token", true),
+    APPLICATION_CERTIFICATE_X5C("Application Certificate SendX5c", true),
+    APPLICATION_NAME_FOR_TRACING("Application Name for Tracing", true),
+    USER_NAME_FOR_TRACING("User Name for Tracing", true),
+    PASSWORD("Password", false), // TODO - should this be enabled for alignment?
+    APPLICATION_CERTIFICATE_BLOB("Application Certificate Blob", false), // TODO - if we add a way to pass a certificate by a byte[], support this
+    APPLICATION_CERTIFICATE_THUMBPRINT("Application Certificate Thumbprint", false),
+    DSTS_FEDERATED_SECURITY("dSTS Federated Security", false),
+    STREAMING("Streaming", false),
+    UNCOMPRESSED("Uncompressed", false),
```

Comment:
what is uncompressed?

Reply:
It's a keyword in c#:

>             Uncompressed, // False by default. If set to true forces the client to not emit the Accept-Encoding header which allows the service to compress the response


We aren't using it anywhere else, it should just know that it's unsupported in case someone copy-pastes from .net